### DISCUSSION TITLE
Scaling difference in impulse response compared to scipy.signal

### DISCUSSION THREAD

**Discussion by fhchl** [python-control/python-control#747] [Q&A] [answered]
I had some code that I wanted to port from using `scipy.signal.StateSpace` etc. to `python-control`. There seems to be some scaling difference between the two implementations. What am I overlooking?

```python
import numpy as np
import control
import scipy.signal

# random state space matrices
t= np.linspace(0, 10, dtype=float)
ss = control.drss(3, 2, 1)
A, B, C, D = ss.A, ss.B, ss.C, ss.D
dt = t[1]

# compute impulse response with python-control
ss = control.ss(A, B, C, D, dt=dt)
t_control, h_control = control.impulse_response(ss, T=t)
plt.figure()
plt.plot(t_control, h_control[0, 0])
plt.plot(t_control, h_control[1, 0])

# compute impulse response with scipy.signal
ss_scipy = scipy.signal.StateSpace(A, B, C, D, dt=dt)
t_scipy, (h_scipy,) = scipy.signal.dimpulse(ss_scipy, t=t)
plt.figure()
plt.plot(t_scipy, h_scipy[:, 0])
plt.plot(t_scipy, h_scipy[:, 1])

plt.show()
```

Creates these two here.

![output](https://user-images.githubusercontent.com/6132231/176428542-02ded514-57bb-48ff-b7c2-902069f55e6c.png)

![output](https://user-images.githubusercontent.com/6132231/176428644-f5295a57-3519-41d4-9c9b-32f302789432.png)

**Comment by bnavigator** [comment, marked as answer]
python-control computes the response to a unit area impulse, rather than a unit height impulse: https://github.com/python-control/python-control/pull/447

**Reply by fhchl** [reply]
That's a subtle and important detail.Thanks! Would it make sense to document this in impulse_response's docstring or am I just too ignorant of this common concept?

**Reply by bnavigator** [reply]
I think it would totally make sense to mention it in the docs.
